address/address_test.go

```diff
@@ -1,159 +1,104 @@
 package address
 
 import (
-	"encoding/json"
-	"fmt"
+	"math/rand"
 	"testing"
+	"time"
 
-	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/assert"
-)
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/require"
+	logging "gx/ipfs/QmbkT7eMTyXfpeyB3ZMxxcxg7XH8t6uXp49jqzz4HB7BGF/go-log"
 
-var hashes = make([][]byte, 5)
+	"github.com/filecoin-project/go-filecoin/bls-signatures"
+	"github.com/filecoin-project/go-filecoin/crypto"
+)
 
 func init() {
-	for i := range hashes {
-		hashes[i] = Hash([]byte(fmt.Sprintf("foo-%d", i)))
-	}
+	logging.SetDebugLogging()
+	rand.Seed(time.Now().Unix())
 }
 
-func TestEmptyAddress(t *testing.T) {
-	assert := assert.New(t)
-	assert.True((Address{}).Empty())
-	assert.False(NewMainnet(hashes[0]).Empty())
-}
+func TestIDAddress(t *testing.T) {
+	require := require.New(t)
 
-func TestNew(t *testing.T) {
-	assert := assert.New(t)
+	addr := NewIDAddress(uint64(rand.Int()))
+	require.Equal(ID, addr.Protocol())
 
-	a := NewMainnet(hashes[0])
-	assert.Len(a.Hash(), 20)
-	assert.Equal(a.Hash(), hashes[0])
+	str, err := encode(Mainnet, addr)
+	require.NoError(err)
 
-	assert.Equal(a.Network(), Mainnet)
-	assert.Equal(a.Version(), uint8(0))
-	assert.Len(a.String(), 41)
-}
+	maybe, err := decode(str)
+	require.NoError(err)
+	require.Equal(addr, maybe)
 
-func TestValidAddresses(t *testing.T) {
-	testCases := []struct {
-		input  Address
-		output string
-	}{
-		{NewMainnet(hashes[0]), "fcqeutlg2sl9daptdcfm8sw7m3xzd0tqhz8f4nzc9"},
-		{NewMainnet(hashes[1]), "fcqwfptkd8ax6xqg7tvycd9wkfyg748fqjwnlt9a0"},
-		{NewTestnet(hashes[0]), "tfqeutlg2sl9daptdcfm8sw7m3xzd0tqhz8g9f95l"},
-		{NewTestnet(hashes[1]), "tfqwfptkd8ax6xqg7tvycd9wkfyg748fqjwj03z34"},
-	}
-
-	for _, tc := range testCases {
-		t.Run(tc.input.String(), func(t *testing.T) {
-			assert := assert.New(t)
-			assert.Equal(tc.input.String(), tc.output)
-
-			a, err := NewFromString(tc.input.String())
-			assert.NoError(err)
-			assert.Equal(a, tc.input)
-
-			assert.NoError(ParseError(a.String()))
-			assert.NoError(ParseError(tc.output))
-			assert.NoError(ParseError(tc.input.String()))
-		})
-
-		t.Run(fmt.Sprintf("roundtrip bytes: %s", tc.input), func(t *testing.T) {
-			assert := assert.New(t)
-
-			a, err := NewFromBytes(tc.input.Bytes())
-			assert.NoError(err)
-			assert.Equal(a, tc.input)
-		})
-	}
 }
 
-func TestInvalidAddressCreation(t *testing.T) {
```

Comment:
This test is still useful for secp256kAddresses. In general, test coverage has gone way down with this PR (more branches, fewer tests).